src/utils/inversePatch.ts

```diff
@@ -0,0 +1,110 @@
+import {
+    AddOperation,
+    CopyOperation,
+    MoveOperation,
+    Operation,
+    RemoveOperation,
+    ReplaceOperation, TestOperation
+} from "fast-json-patch";
+
+// Adapted from https://github.com/cujojs/jiff
+
+const inverses = {
+    test: invertTest,
+    add: invertAdd,
+    replace: invertReplace,
+    remove: invertRemove,
+    move: invertMove,
+    copy: invertCopy
+};
+
+export function inverseOperation(p: Operation[]) {
+    let pr: Operation[] = [];
```

Comment:
I realise the code this is based on looks like this too, but I find this unnecessarily hard to follow for lack of useful variable names.

Reply:
Completely agree, I didn't really want to change this code at all but maybe it would be best since it's _fairly_ short